internal/validations/observability/inputs/application.go

```diff
@@ -0,0 +1,57 @@
+package inputs
+
+import (
+	"fmt"
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	internalobs "github.com/openshift/cluster-logging-operator/internal/api/observability"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"regexp"
+)
+
+var (
+	globRE = regexp.MustCompile(`^[a-zA-Z0-9\*\-]*$`)
+
+	globErrorFmt = `input %q: invalid glob for %s. Must match '` + globRE.String() + `"`
+)
+
+func ValidateApplication(spec obs.InputSpec) []metav1.Condition {
+	if spec.Type != obs.InputTypeApplication {
+		return nil
+	}
+	newCond := func(reason, message string, args ...any) metav1.Condition {
+		if len(args) > 1 {
+			message = fmt.Sprintf(message, args...)
+		}
+		return internalobs.NewCondition(obs.ValidationCondition,
+			metav1.ConditionTrue,
+			reason,
+			message,
+		)
+	}
+
+	if spec.Application == nil {
```

Comment:
this behavior not agreed with `//+kubebuilder:validation:Optional` in `InputSpec`

Reply:
@vparfonov i don't understand what you are expressing. AFAIK there is no way to require a type and enforce the associated spec